libcudacxx/include/cuda/std/__cccl/compiler.h

```diff
@@ -11,25 +11,31 @@
 #ifndef __CCCL_COMPILER_H
 #define __CCCL_COMPILER_H
 
-// Determine the host compiler
+// Determine the host compiler and its version
 #if defined(__INTEL_LLVM_COMPILER)
 #  define _CCCL_COMPILER_ICC_LLVM
+#  define _CCCL_COMPILER_ICC_LLVM_VERSION __INTEL_LLVM_COMPILER
 #elif defined(__INTEL_COMPILER)
 #  define _CCCL_COMPILER_ICC
 #elif defined(__NVCOMPILER)
 #  define _CCCL_COMPILER_NVHPC
+#  define _CCCL_COMPILER_NVHPC_VERSION \
+    (__NVCOMPILER_MAJOR__ * 10000 + __NVCOMPILER_MINOR__ * 100 + __NVCOMPILER_PATCHLEVEL__)
+static_assert(__NVCOMPILER_PATCHLEVEL__ < 100, "__NVCOMPILER_PATCHLEVEL__ must be less than 100");
```

Comment:
That assert will fail with development builds of `nvc++`.  In the dev builds, `__NVCOMPILER_PATCHLEVEL__` is either a date or a Perforce changelist number, both of which are >100.  This will make CCCL completely unusable by `nvc++` developers.

A better assertion would be:
```cpp
static_assert(__NVCOMPILER_MAJOR__ == 99 || __NVCOMPILER_PATCHLEVEL__ < 100,
              "__NVCOMPILER_PATCHLEVEL__ must be less than 100");
```
That verifies the patch number only for release builds.  Dev builds have `__NVCOMPILER_MAJOR__` and `__NVCOMPILER_MINOR__` both set to 99.

Reply:
thanks for the clarification, @dkolsen-pgi